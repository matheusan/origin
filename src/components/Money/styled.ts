import styled from 'styled-components';

export const Money: JSX.Element = styled.div`
  color: ${props => props.theme.colors.brandColorSecondary}
  font-family: ${props => props.theme.fonts.secondary};
  font-size: 32px;
  font-weight: 500;
  text-align: right;

  @media only screen and (max-width: 360px) {
    font-size: 24px;
  }
`;
