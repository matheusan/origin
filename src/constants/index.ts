export const THOUSAND_SEPARATOR = ',';
export const DECIMAL_SEPARATOR = '.';
export const DATE_FORMAT = 'MMMM yyyy';
export const DATE_COMPACT_FORMAT = 'MMM yyyy';
