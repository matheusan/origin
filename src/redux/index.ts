import { SavingsProvider, useSavings } from './context';
import { setAmount, addMonth, subtractMonth, save, saved } from './actions';
import * as selectors from './selectors';

export {
  useSavings,
  setAmount,
  addMonth,
  subtractMonth,
  save,
  saved,
  selectors
};
export default SavingsProvider;
