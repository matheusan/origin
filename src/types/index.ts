export enum TextAlign {
  right = 'right',
  left = 'left',
  center = 'center'
}

export type Goal = {
  id: string;
  icon: string;
  title: string;
  amount?: number;
  date?: number;
};
