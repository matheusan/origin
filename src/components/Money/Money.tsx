import * as React from 'react';

import * as S from './styled';
import { formatMoney } from '../../utils/formatMoney';
import { THOUSAND_SEPARATOR, DECIMAL_SEPARATOR } from '../../constants';
import { TextAlign } from '../../types';

export interface MoneyProps {
  amount: number;
  round?: boolean;
  fontSize?: number;
  textAlign?: TextAlign;
}

const Moeny: React.SFC<MoneyProps> = ({
  amount = 0,
  round = false,
  textAlign = TextAlign.right,
  fontSize
}): JSX.Element => {
  return (
    <S.Money fontSize={fontSize} textAlign={textAlign}>
      ${formatMoney(amount, THOUSAND_SEPARATOR, DECIMAL_SEPARATOR, round)}
    </S.Money>
  );
};

export default Moeny;
