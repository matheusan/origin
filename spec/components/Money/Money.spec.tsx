import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import Money from '../../../src/components/Money';
import { TextAlign } from '../../../src/types';

describe('Money', () => {
  let wrapper: ShallowWrapper;

  it('renders', () => {
    wrapper = shallow(<Money amount={12345.67} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render value as $0.00', () => {
    wrapper = shallow(<Money />);
    expect(wrapper.text()).toBe('$0.00');
  });

  [
    { amount: 2.5, output: '$2.50' },
    { amount: 2.05, output: '$2.05' },
    { amount: 0.07, output: '$0.07' },
    { amount: 299, output: '$299.00' },
    { amount: 1233.0, output: '$1,233.00' },
    { amount: 12345678.1, output: '$12,345,678.10' }
  ].forEach(({ amount, output }) => {
    it(`should render value as ${output}`, () => {
      wrapper = shallow(<Money amount={amount} />);
      expect(wrapper.text()).toBe(output);
    });
  });

  it('should render round amount', () => {
    wrapper = shallow(<Money amount={12345.67} round />);
    expect(wrapper.text()).toBe('$12,345');
  });

  it('should align text to center', () => {
    wrapper = shallow(
      <Money amount={12345.67} round textAlign={TextAlign.center} />
    );
    const styled = wrapper.dive();
    expect(styled.props().textAlign).toBe('center');
  });
});
