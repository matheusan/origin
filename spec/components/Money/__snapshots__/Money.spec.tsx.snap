// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Money renders 1`] = `
<styled.div
  textAlign="right"
>
  $
  12,345.67
</styled.div>
`;
